### ISSUE TITLE
Glyph name confusion

### ISSUE THREAD

**Issue by kilobyte** [fontforge/fontforge#2920]
When trying to build [3270font](https://github.com/rbanffy/3270font) with new fontforge, there's a regression wrt glyph names, and a validation failure.
Because of spurious dependencies, it's easiest to try it from my tree:
```
git clone git@github.com:kilobyte/3270font.git;git checkout 4978e16;make test
```
Fontforge versions 20120731 and 20160404 worked ok, 20161005 interprets "Omega" as deprecated U+2126 OHM SIGN rather than regular U+03A9 GREEK CAPITAL LETTER OMEGA.
```
Attempt to unget two characters
CHECKING     3270Medium.otf
The glyph named Omega is mapped to U+03A9.
  But its name indicates it should be mapped to U+2126.
PASS         3270Medium.otf

Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 20161015-ML-D.
 Based on source from git with hash: 
Attempt to unget two characters
CHECKING     3270Medium.pfm
ERROR     22 Two glyphs have the same Unicode code point
FAIL         3270Medium.pfm
```
In this particular case, there's an easy workaround: explicitly writing Omega.greek instead of Omega makes the failure go away, but I'm reporting in case this affects more fonts that are harder to change.

**Comment by frank-trampe** [comment]
@JoesCat, this looks like an encoding issue.

**Comment by JoesCat** [comment]
Hi @kilobyte,
Whether this bug is fixed or not, I do recommend that you do use the longer more detailed definitions, such as Omega.greek, and not the shorter definitions to avoid these sorts of potential bugs (Upstream bug here: https://github.com/rbanffy/3270font/pull/25 ). I'm +1 on recommending the extra wording.

Doing a grep and searching for terms such as "3A9", "2126" shows some fluidity within FontForge (example: https://github.com/fontforge/fontforge/commit/20b9ce1cbcfdc7d620b2b11ab5f7f075987fbad4 ), as well as with upstream charts with new characters introduced and others deprecated.
Looking at other tables such as Tex encoding or other spots with variables like ".1992", ".1980", suggests we may need to add another variable, like YEAR or VERSION when it comes to import/export

Thanks for the recommendation of follow-up using your git tree, this should help narrow it down.

**Comment by kilobyte** [comment]
Right, so it's an issue of glyph names from sources other than Unicode.  Would you then recommend changing all other glyphs in the greek (and possibly cyrillic/etc) range, or shouldn't we bother?

**Comment by JoesCat** [comment]
Looking at UnicodeData{6.3/9.0}.txt, NamesList{6.3/9.0}.txt, PropList{6.3,9.0}.txt, there are inconsistencies that may change, like the Ohm symbol==U2126, and KOhm==K+U03A9.
In 8859-16, some characters/ligatures, were made using a substitute keyboard/language, so it might change.
I've watched some programmers scratching their heads because they used the assumptions of routines self-initialize with zero, or carry "default" values, then run into problems as their program got more complex. The extra definition could help remove assumption, but you don't need to do every character since there are only a few types with interesting characteristics like the Greek, Math, Electrical Omega.

**Comment by JoesCat** [comment]
....Problem at the moment seems to be after 20160404 but before 20160430

**Comment by JoesCat** [comment]
Hi @kilobyte,
I've been trying to understand this issue a little better, and what it seem to me is that this is more of a glyph-line intersection problem with the ttf results causing your 'make test' to come to a hard stop.

Going back to 20160404, your test continues with the 3rd fault (ttf) shown below, then continues onwards until the end of the tests. If I run HEAD, it hard-stops visibly at the 2nd problem, but I think it is stopping at the ttf intersection problem and not displaying a result. If I try an earlier FontForge (20160824, 20141014), I was expecting to see corrected Omega statements, but I see the same message and think this goes back to the ttf intersection again.
```txt
Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 20:03 PDT 25-Oct-2016-ML-D.
 Based on source from git with hash: 94038cd93f463ae5020b722a518c65abc843a79a
Attempt to unget two characters
CHECKING     3270Medium.otf
The glyph named Omega is mapped to U+03A9.
  But its name indicates it should be mapped to U+2126.
PASS         3270Medium.otf

Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 20:03 PDT 25-Oct-2016-ML-D.
 Based on source from git with hash: 94038cd93f463ae5020b722a518c65abc843a79a
Attempt to unget two characters
CHECKING     3270Medium.pfm
ERROR     22 Two glyphs have the same Unicode code point
FAIL         3270Medium.pfm

Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 20:03 PDT 25-Oct-2016-ML-D.
 Based on source from git with hash: 94038cd93f463ae5020b722a518c65abc843a79a
Attempt to unget two characters
CHECKING     3270Medium.ttf
The glyph named Omega is mapped to U+03A9.
  But its name indicates it should be mapped to U+2126.
ERROR      2 Self-intersecting glyph
FAIL         3270Medium.ttf
```
I'm just pointing this out since I note the statements referring to U+03A9/U+2126 are the same for 20160404.

Following the "Omega" naming problem, I note that your fonts refer to "Omega" and "omega".
...plus, haven't found out how the tables in namelist.c were created...might be handmade.
...plus I think there might be a hand-modified edit at 2006-10-02 13:55:57 "Apple updated the defn. of MacRoman a while ago and the Omega-like symbol has changed from being an Ohm sign to being an Omega. Alexej Kryukov seems to care."

....still needs more investigation...
